packages/react-client/src/hooks/devices/useInitializeDevices.ts

```diff
@@ -1,72 +1,85 @@
-    async ({ enableVideo = true, enableAudio = true }: UseInitializeDevicesParams = {}) => {
-      if (hasDevicesBeenInitializedRef.current) return;
-      hasDevicesBeenInitializedRef.current = true;
+  const initializeDevices: (params?: UseInitializeDevicesParams) => Promise<null | InitializeDevicesErrors> =
+    useCallback(
+      async ({ enableVideo = true, enableAudio = true }: UseInitializeDevicesParams = {}) => {
+        if (hasDevicesBeenInitializedRef.current) return null;
+        hasDevicesBeenInitializedRef.current = true;
+
+        const videoManager = videoDeviceManagerRef.current;
+        const audioManager = audioDeviceManagerRef.current;
 
-      const videoManager = videoDeviceManagerRef.current;
-      const audioManager = audioDeviceManagerRef.current;
+        const constraints = {
+          video: enableVideo && videoManager.getConstraints(),
+          audio: enableAudio && audioManager.getConstraints(),
+        };
 
-      const constraints = {
-        video: enableVideo && videoManager.getConstraints(),
-        audio: enableAudio && audioManager.getConstraints(),
-      };
+        const previousDevices = {
+          video: videoManager.getLastDevice(),
+          audio: audioManager.getLastDevice(),
+        };
 
-      const previousDevices = {
-        video: videoManager.getLastDevice(),
-        audio: audioManager.getLastDevice(),
-      };
+        // Attempt to start the last selected device to avoid an unnecessary restart.
+        // Without this, the first device will start, and `correctDevicesOnSafari` will attempt to fix it.
+        let [stream, deviceErrors] = await getAvailableMedia({
+          video: enableVideo && prepareConstraints(previousDevices.video?.deviceId, constraints.video),
+          audio: enableAudio && prepareConstraints(previousDevices.audio?.deviceId, constraints.audio),
+        });
 
-      // Attempt to start the last selected device to avoid an unnecessary restart.
-      // Without this, the first device will start, and `getCorrectedResult` will attempt to fix it.
-      let [stream, deviceErrors] = await getAvailableMedia({
-        video: enableVideo && prepareConstraints(previousDevices.video?.deviceId, constraints.video),
-        audio: enableAudio && prepareConstraints(previousDevices.audio?.deviceId, constraints.audio),
-      });
+        const devices = await navigator.mediaDevices.enumerateDevices();
 
-      const devices = await navigator.mediaDevices.enumerateDevices();
+        const videoDevices = devices.filter(({ kind }) => kind === "videoinput");
+        const audioDevices = devices.filter(({ kind }) => kind === "audioinput");
 
-      const videoDevices = devices.filter(({ kind }) => kind === "videoinput");
-      const audioDevices = devices.filter(({ kind }) => kind === "audioinput");
+        if (stream) {
+          [stream, deviceErrors] = await correctDevicesOnSafari(
```

Comment:
Why calling this method is in responsibility of `useInitializeDevices`? Should this be part of `getAvailableMedia` method? (or other method exported from `mediaInitializer`, so  `getAvailableMedia` and `correctDevicesOnSafari` won't have be called separately here)

Reply:
i played around a bit and decided this is worth of another ticket